mediator/med_phases_aofluxes_mod.F90

```diff
@@ -563,13 +1259,138 @@ subroutine med_aofluxes_run(gcomp, aoflux, rc)
          duu10n=aoflux%duu10n, ustar_sv=aoflux%ustar, re_sv=aoflux%re, ssq_sv=aoflux%ssq, &
          missval = 0.0_r8)
 
-    do n = 1,lsize
+    do n = 1,aoflux%lsize
        if (aoflux%mask(n) /= 0) then
           aoflux%u10(n) = sqrt(aoflux%duu10n(n))
        end if
     enddo
+
+    !----------------------------------
+    ! map aoflux to output to relevant atm/ocn grid(s)
+    !----------------------------------
+
+    if (is_local%wrap%aoflux_grid == 'ogrid') then
+
+       ! mapping aoflux from ogrid to agrid is done in med_phases_prep_atm using updated ocean fractions
+       ! on the atm grid
+
+    else if (is_local%wrap%aoflux_grid == 'agrid') then
+
+       if (is_local%wrap%med_coupling_active(compatm,compocn)) then
+          ! map aoflux from agrid to ogrid
+          do nf = 1,size(fldnames_aof_out)
+             ! Create source field
+             call ESMF_FieldBundleGet(is_local%wrap%FBMed_aoflux_a, fldnames_aof_out(nf), field=field_src, rc=rc)
+             if (chkerr(rc,__LINE__,u_FILE_u)) return
+             ! Create destination field
+             call ESMF_FieldBundleGet(is_local%wrap%FBMed_aoflux_o, fldnames_aof_out(nf), field=field_dst, rc=rc)
+             if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+             ! Map atm->ocn conservatively WITHOUT fractions
+             call ESMF_FieldRegrid(field_src, field_dst, &
+                                !routehandle=is_local%wrap%RH(compatm, compocn, mapconsd), &
+                  routehandle=is_local%wrap%RH(compatm, compocn, mapconsf), &
+                  termorderflag=ESMF_TERMORDER_SRCSEQ, zeroregion=ESMF_REGION_TOTAL, rc=rc)
+
+             ! One normalization
```

Comment:
Should this code block be removed?

Reply:
Why do you think this code block needs to be removed. Its what maps the output aoflux from the atm grid to the ocn grid. Am I missing something in our comment?

Reply:
refering to lines 1295 - 1306